### PR TITLE
Decouple Pooler from UniswapV2 use case

### PR DESCRIPTION
**Pull request by xadahiya**
### Checklist
- [x] My branch is up-to-date with upstream/develop branch.
- [x] Everything works and tested for Python 3.8.0 and above.
- [x] I ran pre-commit checks against my changes.

### Current behavior

The current version of Pooler is very tightly coupled with the UniswapV2 use case, which makes developing new use cases using pooler a time-consuming and complicated process. 
### New expected behavior

Pooler should efficiently enable the development of new use cases. The framework should take care of most of the heavy lifting allowing developers to focus on just their use case logic.

### Change logs
This PR aims to decouple Pooler from the UniswapV2 use case and make use case development easy, fast, and modular. Additionally, it introduces some helper utilities and guides to make the use case development cycle faster.

#### Added
- [RpcHelper](https://github.com/PowerLoom/pooler/blob/pooler_decoupled/pooler/utils/rpc.py#L90) to efficiently interact with blockchain
- [GenericProcessor](https://github.com/PowerLoom/pooler/blob/pooler_decoupled/pooler/utils/callback_helpers.py#L128) to build callback workers for custom use-case development
- [Project Config](https://github.com/PowerLoom/fpmm-pooler-internal/blob/pooler_decoupled/pooler/static/projects.example.json) for plug and play use case architecture

#### Detailed Changes
- remove hardcoding of `cached_pairs.json` and dependency on it
- `PairTotalReservesProcessorDistributor` is now `ProcessorDistributor` managed by process_core_hub
- Cleanup messages broadcasted by `EpochCallbackManager`
- Cleanup Queue, remove hardcoded names and subtopic feature
- Remove unused diff-related functions from `AuditProtocolCommandsHelper`
- Remove `module_queues_config.json`, everything controlled by projects.json now
- Refactor pooler `settings.json` and move worker specific config to callback_modules `settings.json`
- Pair reserves and Trade volume calculation moved to separate dynamic workers
- Move `_send_audit_payload_commit_service` to CallbackAsyncWorker after generalization
- Cleanup and segregate redis_keys
- Fix code quality and style issues. Only type annotation issues remaining
- Single RPC interface implementation
- Remove dependency of gnosis-py

## Deployment Instructions
Refer [here](https://github.com/PowerLoom/pooler/blob/pooler_decoupled/README.md#development-instructions)

## Building Use Cases using pooler
Refer [here](https://github.com/PowerLoom/pooler/blob/pooler_decoupled/README.md#building-usecases-using-pooler)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -71,20 +50,20 @@ These instructions are needed if you're planning to run the system using `build-
...
+#### Configuring `pooler/settings/settings.json`
+There can be a lot to fine tune in `settings.json` but the following are essential.
 
-- `instance_id`, it is currently generated on invite only basis (refer [deploy](https://github.com/PowerLoom/deploy) repo for more details)
+- `instance_id`: This is the unique identifier for your node to participate in consensus. It is currently generated on invite only basis (refer [deploy](https://github.com/PowerLoom/deploy) repo for more details on applying for an instance ID).
 - `namespace`, it is the unique key used to identify your project namespace
-- rpc url and rate limit config in `rpc.full_nodes`, this depends on which rpc node you're using
-- consensus url in `consensus.url`, this the the offchain-consensus service url where snapshots will be submitted
+- RPC service URL(s) and rate limit configurations against them  in `rpc.full_nodes`. This will correspond to RPC nodes or service providers for the chain on which the data source smart contracts live (for eg. Ethereum Mainnet, Polygon Mainnet etc)
+- Consensus URL in `consensus.url`, this the the offchain consensus service to which snapshots will be submitted
```
Comment (anomit) [inline README.md:63]: Typo: `this the the`

Comment (anomit) [reply README.md:63]: We should mention the usage of a rate limiting library that works on local Redis cache to rate limit RPC calls according to the plan subscribers have with their providers.

And maybe a line on the importance of running our own nodes wherever possible?

Comment (xadahiya) [reply README.md]: Updated!

**Review by anomit — APPROVED** [review]
Approving since data has been observed to be accurate across multiple staging systems for aggregated datapoints of liquidity, 24 hours trade volume and fees

**Comment by xadahiya** [comment]
Superseded by https://github.com/PowerLoom/pooler/pull/25, closing

**Pull request closed without merging**